### PR TITLE
Add support for the selection of child item in the table

### PR DESCRIPTION
**Pull request by rwx788**
We got new feature for the YTable, so that items in the table can have
nested items, which we were not able to select previously. This change
extends REST API functionality to accept path to the item in the table,
e.g. `root_row|sub_item_row1|sub_item_row2`.

This functionality is currently used in the expert partitioner.

### CODE REVIEW

File: src/YTableActionHandler.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+#define YUILogComponent "rest-api"
+#include "YUILog.h"
+
+std::function<void (YTable*)> YTableActionHandler::get_handler( YTable * widget,
+                                                                const std::string &value,
+                                                                const int &column_id,
+                                                                const int &row_id )
+{
+    // Handle case when want select row by row number
```
Comment (shundhammer) [inline src/YTableActionHandler.cc]: when _we_ want _to_ select  _a_ row by _the_ row number  ?

Comment (rwx788) [reply src/YTableActionHandler.cc]: Improved that comment, thanks!

File: src/YTableActionHandler.cc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                                     YItemConstIterator                       begin,
+                                     YItemConstIterator                       end,
+                                     const int                               &column_id )
+{
+    for ( YItemConstIterator it = begin; it != end; ++it )
+    {
+        YTableItem * item = dynamic_cast<YTableItem *>( *it );
+
+        if ( ! item )
+            return 0;
```
Comment (lslezak) [inline src/YTableActionHandler.cc]: `nullptr` is a bit cleaner way...

```suggestion
            return nullptr;
```

File: src/YTableActionHandler.cc [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                                                                       item->childrenBegin(),
+                                                                       item->childrenEnd(),
+                                                                       column_id );
+
+            if ( result )
+                return result;
+        }
+    }
+
+    return 0;
```
Comment (lslezak) [inline src/YTableActionHandler.cc]:
```suggestion
    return nullptr;
```

File: src/YTableActionHandler.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+class YTableActionHandler
+{
+public:
+
+    YTableActionHandler() {}
+    virtual ~YTableActionHandler() {}
+
+    static std::function<void (YTable*)> get_handler( YTable * widget,
+                                                      const std::string &value,
+                                                      const int &column_id = 0,
```
Comment (lslezak) [inline src/YTableActionHandler.h]: Could you please add a comment for these "magic" default values? What do they mean? It's a bit confusing why `0` and `-1` are used, esp. why here there is `column_id = 0` and in the function below there is `column_id = -1` default.

**Review by lslezak — COMMENTED** [review]
Just prefer `nullptr` and comment the default values

File: src/YTableActionHandler.h (hunk 4 above)
Comment (rwx788) [reply src/YTableActionHandler.h]: Yeah, it's confusing as `0` is used by default in libyui parent library, and `-1` will mean that we will get default value. But we should be consistent, so I will take care of that.

File: src/YTableActionHandler.cc (hunk 3 above)
Comment (rwx788) [reply src/YTableActionHandler.cc]: Done.

File: src/YTableActionHandler.cc (hunk 2 above)
Comment (rwx788) [reply src/YTableActionHandler.cc]: Makes total sense, done.

File: src/YTableActionHandler.h (hunk 4 above)
Comment (rwx788) [reply src/YTableActionHandler.h]: Point taken, thanks!

**Review by lslezak — APPROVED** [review]

**Pull request merged**